File: sample-projects/c/src/domain/model/centaur.c
#include <stdlib.h>

#include "../../../include/de/sots/cellarsandcentaurs/domain/model/centaur.h"
#include "dice_table.c"

typedef struct CentaurCharge {
    DiceTable table;
    int bonus;
} CentaurCharge;

Centaur *centaur_new(CreatureId id)
{
    Centaur *centaur = calloc(1, sizeof(Centaur));
    centaur->base.id = id;
    centaur->bow_range = dice_table_max(&DICE_TABLE_D20);
    return centaur;
}

int centaur_charge(Centaur *centaur, Creature *target)
{
    CentaurCharge charge = { DICE_TABLE_D20, centaur->bow_range / 10 };
    int damage = charge.table.average + charge.bonus;
    target->hit_points.current -= damage;
    return damage;
}

File: sample-projects/go/domain/model/centaur.go
package model

type Centaur struct {
	Creature
	XPValue int
}

func NewCentaur(id CreatureId) *Centaur {
	centaur := &Centaur{Creature: *NewCreatureOfType(id, Monstrosity), XPValue: 450}
	centaur.SetArmorClass(NewArmorClass(12, 0))
	return centaur
}

func (c *Centaur) Charge() Speed {
	return c.Speeds()[Walking]
}
